apple/build_settings/build_settings.bzl

```diff
@@ -38,6 +38,12 @@ Enables Bazel's tree artifacts for Apple bundle rules (instead of archives).
 """,
         default = False,
     ),
+    "use_library_evolution": struct(
+        doc = """
+Enables library evolution, allowing for the emission of .swiftinterface files.
+""",
+        default = True,
+    ),
```

Comment:
How does this interact with the `apple.no_legacy_swiftinterface`, `swift.emit_swiftinterface`, and `swift.enable_library_evolution` features?

Reply:
As for `swift.emit_swiftinterface` and `swift.enable_library_evolution`, when `use_library_evolution` has a value of False, they will also be set to `False` for all transitive dependencies.

As for `apple.no_legacy_swiftinterface` (I understand you mean the one declared here https://github.com/bazelbuild/rules_apple/pull/2104) - the behavior is basically the same. From my point of view, it is more convenient to use the flag in the scope of the whole build. However, I am totally fine to use `apple.no_legacy_swiftinterface`.

Reply:
Do you prefer any of the approaches? Would it make sense to cherry pick `apple.no_legacy_swiftinterface` into this PR (or merge it into `main`)?